### PR TITLE
Add custom css custom properties for background-color and color. Update docs.

### LINKED ISSUE #7: Need summary docs
No docs at all here: https://github.com/PolymerElements/paper-toast/blob/56ff0143cf409a81d505dfa744f713dee68c5647/paper-toast.html

### LINKED ISSUE #18: Add --paper-toast CSS variable
There's no way to style this element from outside of element. Please add `--paper-toast` CSS variable.

### LINKED ISSUE #41: Example source missing from the documentation page
On the https://elements.polymer-project.org/elements/paper-toast site the documentation missing an example usage source like the other elements has. Please add something like this:

```
<paper-toast id="example_toast" text="Some message in toast!"></paper-toast>

<script>
    document.getElementById('example_toast').show();
</script>
```

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #18 by providing 2 new css custom properties to style the toast.
Fixes #7, fixes #41 by providing documentation for properties, events and summary of the element.

Demo page updated, and some code cleanup.

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -39,20 +39,43 @@
...
 
-  <paper-button raised onclick="document.querySelector('#toast1').show()">Discard Draft</paper-button>
+  <paper-button raised onclick="openToast('#toast1')">Discard Draft</paper-button>
+
+  <paper-button raised onclick="openToast('#toast2')">Get Messages</paper-button>
+
+  <paper-button raised onclick="openToast('#toast3')">Get an error</paper-button>
 
-  <paper-button raised onclick="document.querySelector('#toast2').show()">Get Messages</paper-button>
+  <paper-toast id="toast0" class="fit-bottom" text="Hi there! This toast will stay opened until you open another toast." opened duration="0"></paper-toast>
```
Comment (cdata) [inline demo/index.html]: :+1:

File: demo/index.html [hunk 2]
```diff
@@ -39,20 +39,43 @@
...
 
   <paper-toast id="toast1" text="Your draft has been discarded."></paper-toast>
 
-  <paper-toast id="toast2" text="Connection timed out. Showing limited messages." duration="0" onclick="this.hide()">
-    <span role="button" tabindex="0" style="color: #eeff41;margin: 10px" onclick="console.log('RETRY')">Retry</span>
```
Comment (cdata) [inline demo/index.html]: :100:

File: paper-toast.html [hunk 3]
```diff
@@ -30,9 +51,8 @@
       :host {
         display: block;
         position: fixed;
-
-        background: #323232;
-        color: #f1f1f1;
+        background-color: var(--paper-toast-background-color, #323232);
+        color: var(--paper-toast-color, #f1f1f1);
```
Comment (cdata) [inline paper-toast.html]: :beers:

File: paper-toast.html [hunk 4]
```diff
@@ -18,6 +18,27 @@
 
 `paper-toast` provides a subtle notification toast.
 
+Example:
+
+    <paper-toast text="Welcome back!"></paper-toast>
+
+Set `duration` to 0, a negative number or Infinity to persist the toast on screen
```
Comment (cdata) [inline paper-toast.html]: :+1:

File: paper-toast.html [hunk 5]
```diff
@@ -40,17 +60,15 @@
...
         margin: 12px;
         font-size: 14px;
         cursor: default;
-        -webkit-transition: visibility 0.3s, -webkit-transform 0.3s, opacity 0.3s;
-        transition: visibility 0.3s, transform 0.3s, opacity 0.3s;
-
-        visibility: hidden;
-        bottom: -100px;
+        -webkit-transition: -webkit-transform 0.3s, opacity 0.3s;
+        transition: transform 0.3s, opacity 0.3s;
```
Comment (cdata) [inline paper-toast.html]: Isn't the visibility transition duration important for something? I can't quite remember, but I seem to recall that it affects screen readers somehow..

File: paper-toast.html [hunk 6]
```diff
@@ -111,25 +131,34 @@
           },
         },
 
+        /**
+         * @property visible
+         * @deprecated
```
Comment (cdata) [inline paper-toast.html]: :clap:

File: paper-toast.html [hunk 7]
```diff
@@ -111,25 +131,34 @@
...
+         * Same as `opened`. Deprecated.
+         */
         get visible() {
           console.warn('`visible` is deprecated, use `opened` instead');
           return this.opened;
         },
 
+        // can auto-close if duration is a positive finite number
+        get _canAutoClose() {
```
Comment (cdata) [inline paper-toast.html]: :+1:

File: paper-toast.html [hunk 8]
```diff
@@ -111,25 +131,34 @@
...
+         * @property visible
+         * @deprecated
+         * Same as `opened`. Deprecated.
+         */
         get visible() {
           console.warn('`visible` is deprecated, use `opened` instead');
           return this.opened;
         },
 
+        // can auto-close if duration is a positive finite number
```
Comment (cdata) [inline paper-toast.html]: Please use the documentation style we use for other methods for this. Also, include annotations! :bow:

File: paper-toast.html [hunk 9]
```diff
@@ -139,6 +168,14 @@
...
          * Called when the value of `opened` changes.
          */
+
+        /**
+         * Fired when `paper-toast` is opened.
+         *
+         * @event 'iron-announce'
+         * @param {Object} detail
+         * @param {String} detail.text The text that will be announced.
+         */
```
Comment (cdata) [inline paper-toast.html]: This block makes it look like the documentation is for `_openedChanged`. Usually we put event docs  towards the bottom of the implementation, for example: https://github.com/PolymerElements/paper-slider/blob/master/paper-slider.html#L739-L759

File: paper-toast.html [hunk 10]
```diff
@@ -176,7 +212,7 @@
          * iron-fit-behavior will set the inline style position: static, which
          * causes the toast to be rendered incorrectly when opened by default.
          */
-        _onIronResize: function(){
+        _onIronResize: function() {
```
Comment (cdata) [inline paper-toast.html]: Nice :+1:

**Comment by cdata** [comment]
This is looking pretty good. Just had some minor style feedback. PTAL!

File: paper-toast.html (hunk 6 above)
Comment (valdrinkoshi) [inline paper-toast.html]: Unfortunately this doesn't appear in the docs. I see the `visible` property, but not the description. Am I missing something?

File: paper-toast.html (hunk 5 above)
Comment (valdrinkoshi) [inline paper-toast.html]: I completely removed `visibility`, as now we are using `opacity`. The effect is the same, both these properties keep the element in the document flow. The advantage of using `opacity` is that we it has a nicer, visible animation.

File: paper-toast.html (hunk 8 above)
Comment (valdrinkoshi) [inline paper-toast.html]: ok, will declare it as `@private`

File: paper-toast.html (hunk 9 above)
Comment (valdrinkoshi) [inline paper-toast.html]: roger that, will move it to the bottom!

File: paper-toast.html (hunk 6 above)
Comment (valdrinkoshi) [inline paper-toast.html]: Ok, I figured it out. The description must go on top, before the annotations

**Comment by valdrinkoshi** [comment]
@cdata feedback incorporated!

File: paper-toast.html (hunk 5 above)
Comment (valdrinkoshi) [inline paper-toast.html]: oh, about the screen readers, i didn't consider that! Well, I guess I'll put it back to play it safe..

File: paper-toast.html (hunk 9 above)
Comment (valdrinkoshi) [inline paper-toast.html]: done

**Comment by cdata** [comment]
LGTM :+1:

**Pull request merged**